### ISSUE TITLE
I'm interested in picking this up

### ISSUE THREAD

**Issue by toftat** [LenShustek/RedLINK-Sniffer#2]
I am on the verge of getting some redlink devices and would like to bring them into my home automation setup, in the process of which I came across this stuff.

I have an EZ-430 chronos (see here http://processors.wiki.ti.com/index.php/EZ430-Chronos) which comes with a USB based CC1111 (or CC1101) device that I might repurpose  to manage the communications with the redlink network.

I will need to figure out how to get code onto the device and see what I can make it do, but I suspect that I will be able to get the RF aspects into the microcontroller firmware and provide  a simple text based output.

To you have any advise beyond what is here?

**Comment by LenShustek** [comment]
My frustration was such that I decommissioned all of my RedLINK gear a couple of years ago and replaced it with wireless thermostats that I designed and built myself: https://github.com/LenShustek/tempsensor. 
So unfortunately I don't have any more recent information or advice about reverse-engineering RedLINK.  But out of intellectual curiosity I would be very interested in what you learn! I'd especially like to understand how the receiver synchronizes with multiple transmit-only sensors that are communicating by frequency-hopping.

**Comment by toftat** [comment]
Thanks for the update!

Unfortunately I am fighting a significant WAF problem, so if it's visible it has to look store bought, and it has to work right out of the box. So building and installing my own stuff is a challenge. However if I can install the system and then interpret the data I can do stuff with it.

If _I_ was designing the frequency hopping I would make each transmitted packet the largest it could be, while at the same time fitting into one hop. Then when the time came to send the packet, I would transmit it on the current channel, then hop and retransmit a few times. On the receiver it would hop as fast as possible, listening to each channel. Once it saw a channel in use, it would hop once more and wait for a while. This way when the transmitter does a hop and retransmit cycle, it would transmit on the channel the receiver is now listening to. Once the receiver gets the packet (or has a timeout), it would then return to scanning.

After the transmitter sent it, it would choose another channel at random and when the time came for the next transmit start the process again. 

I would make the next channel and next transmit time be random so that if 2 sensors stepped on each other on _this_ reading they'd be highly unlikely to do it again on the next reading. As long as the frequency of readings are reasonable (2-3 minutes or less between for example) the user won't notice if they occasionally got lost.

Again, that's how I'd do it, not saying anything about how redlink does it.

**Comment by laynd** [comment]
@toftat Any success in your efforts? I got Mitsubishi HVAC system with Redlink adapters. Wondering if anyone had any success with Redlink to Bacnet translation. There is a Redlink to web adapter by Honeywell but I assume it will require a good deal of reverse engineering to get useful data

**Comment by toftat** [comment]
No, I decided that I was not going to dig into this, and explored designing and building my own controller and thermostats, it turns out that the traditional controller is very basic, and emulating it is relatively simple, a handful of relays is all you need, the smarts are important when deciding which relays to close when.

As I was researching this, I came across the Ecobee line that did what I wanted out of the box so I went with that product.

**Comment by KTibow** [comment]
I'm also interested in this. I want to set something up for some of my family members but they don't want me to connect anything to the thermostat or unit, so connecting wirelessly seems like the only option.
